dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -520,6 +551,41 @@ private static boolean containsIgnoreCase(Collection<String> featureToInstall, S
 
             int alreadyInstalled = 0;
             Collection<String> featureToInstall = (Collection<String>) data.get(FEATURES_TO_RESOLVE);
+            System.out.println("features to resolve: " + data.get(FEATURES_TO_RESOLVE));
+            System.out.println("install individual esas: " + data.get(INSTALL_INDIVIDUAL_ESAS));
+            //TODO
+            try {
+                if (data.get(INSTALL_INDIVIDUAL_ESAS).equals(Boolean.TRUE)) {
+                    Path tempDir = Files.createTempDirectory("generatedJson");
+                    tempDir.toFile().deleteOnExit();
+                    Map<String, String> shortNameMap = new HashMap<String, String>();
+                    File individualEsaJson = individualESAInstall(tempDir, shortNameMap);
+                    RepositoryConnection repo = new SingleFileRepositoryConnection(individualEsaJson);
+                    repoList.add(repo);
+                    for (String feature : featureToInstall) {
+                        if (feature.endsWith(".esa")) {
+                            if (shortNameMap.containsKey(feature)) {
```

Comment:
The `shortNameMap` was populated by `individualESAInstall()` with the keys being absolute paths to the ESA files.  But can ESA files be specified (e.g. in the Maven plugin's `<features>` section) with relative paths?  (You can try this with the existing plugin, which will use installUtility for individual esas.)  If so, then the `feature` string here could be a relative path, so `shortNameMap.containsKey(feature)` would be false even though we want it to match.  You could get around this by comparing canonical paths instead (not absolute paths, because absolute paths would not resolve things such as "." or "..")

Reply:
This was tested and ESA files cannot be specified with relative paths with installUtility.

Reply:
What if they have absolute paths but with things such as "." or "..".  For example, `/folder/../folder/file.esa` and `/folder/file.esa` point to the same file, but comparing absolute paths would give false.  So we should still use canonical paths (so that both will be resolved to `/folder/file.esa`)

Reply:
This structure also results in a failed to invoke error.